huggingface_pipelines/speech.py

```diff
@@ -0,0 +1,236 @@
+import os
+import torch
+from typing import Dict, Any
+from dataclasses import dataclass, replace
+from sonar.inference_pipelines.speech import SpeechToEmbeddingModelPipeline
+import tempfile
+import soundfile as sf
+import logging
+from .pipeline import Pipeline, PipelineConfig, PipelineOverwrites
+from .dataset import DatasetConfig
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class HFSpeechToEmbeddingOverwrites(PipelineOverwrites, total=False):
```

Comment:
Let's keep the name consistent, either all "speech" or all "audio" (I prefer the latter as I see none of your implementation special for speech)